Source/CarthageKit/Xcode.swift

```diff
@@ -1012,9 +1012,35 @@ public func stripDSYM(_ dSYMURL: URL, keepingArchitectures: [String]) -> SignalP
 
 /// Strips a universal file from unexpected architectures.
 private func stripBinary(_ binaryURL: URL, keepingArchitectures: [String]) -> SignalProducer<(), CarthageError> {
-	return architecturesInPackage(binaryURL)
-		.filter { !keepingArchitectures.contains($0) }
-		.flatMap(.concat) { stripArchitecture(binaryURL, $0) }
+  let fileManager = FileManager.default.reactive
+  
+  let createTempDir: SignalProducer<URL, CarthageError> = fileManager.createTemporaryDirectoryWithTemplate("carthage-lipo-XXXXXX")
+  
+  let copyItem: (URL, URL) -> SignalProducer<URL, CarthageError> = { source, dest in
+    fileManager.copyItem(source, into: dest)
+  }
+  
+  let strip: (URL, [String]) -> SignalProducer<URL, CarthageError> = { workspace, keeping in
+    return architecturesInPackage(workspace)
+      .filter { !keepingArchitectures.contains($0) }
+      .flatMap(.concat) { stripArchitecture(workspace, $0) }
+      .then(SignalProducer(value: workspace))
+  }
+  
+  let replace: (URL, URL) -> SignalProducer<(), CarthageError> = { original, modified in
+    return fileManager.replaceItem(at: original, withItemAt: modified)
+  }
+  
+  return createTempDir
```

Comment:
Would you mind adding a comment here that explains why we're going through the trouble of copying/stripping/replacing instead of just stripping?

Reply:
Done!